docs/news/posts/2024-10-04-data-access.md

```diff
@@ -0,0 +1,226 @@
+(at least for most of us) easier than writing raw SQL queries, its syntax usually requires some familiarization [^6].
+
+Recent improvements have therefore enabled an alternative, more intuitive way to construct queries. Let us explain with
+the following example: Assume you wanted to obtain all integers with `pk`s in a range between 1 and 10 (both excluded)
+from a `Group` called "integers". You'd have to construct the following, rather convoluted query:
+
+```python
+from aiida import orm
+
+qb = orm.QueryBuilder()
+qb.append(
+    orm.Group,
+    filters={
+        "label": "integers",
+    },
+    project=["label"],
+    tag="group",
+)
+qb.append(
+    orm.Int,
+    with_incoming="group",
+    filters={"and": [{"pk": {">": 1}}, {"pk": {"<": 10}}]},
+    project=["pk", "attributes.value"],
+)
+```
+
+With the new syntax, `Node` `attribute`s are directly accessible via the `.fields` specifier and, as such, the logic can be applied to them directly:
+
+```python
+from aiida import orm
+
+qb = orm.QueryBuilder()
+qb.append(
+    orm.Group,
+    filters=orm.Group.fields.label == "integers",
+    project=[orm.Group.fields.label],
+    tag="group",
+)
+qb.append(
+    orm.Int,
+    with_incoming="group",
+    filters=(orm.Int.fields.pk > 1) & (orm.Int.fields.pk < 10),
+    project=[orm.Int.fields.pk, orm.Int.fields.value],
+)
+```
+
+Through this, for example the filter on the values of the `pk`s reduces from:
+
+```python
+filters={
+    "and": [
+        {"pk": {">": 1}},
+        {"pk": {"<": 10}},
+    ]
+}
+```
+
+to the more concise:
+
+```python
+filters=(orm.Int.fields.pk > 1) & (orm.Int.fields.pk < 10),
+```
+
+in which the `"and"` condition can be expressed via `&` and applied on the relevant entities. Furthermore,
+accessing through the `.fields` attribute, e.g. in the updated `project`ion:
+
+```python
+project=[orm.Int.fields.pk, orm.Int.fields.value]
+```
+
+albeit being more verbose, is less error-prone than the access via string identifiers in the previous version:
+
+```python
+project=["pk", "attributes.value"]
+```
+
+as it allows for autocompletion.
+
+Any feedback on the new QB syntax welcome!
+
+***
+
+## Relevant PRs
+
+For the more tech-savvy among us, here are the relevant PRs of the changes outlined in this blog post:
+
+- [[#6276]](https://github.com/aiidateam/aiida-core/pull/6276) Add CLI command to dump inputs/outputs of `CalcJob`/`WorkChain`
+- [[#6245]](https://github.com/aiidateam/aiida-core/pull/6245) ✨ NEW: Add `orm.Entity.fields` interface for `QueryBuilder`
+  (cont.) [and linked PRs]
+
+## Footnotes
+
+[^1]:
+    The file repository is based on the [`disk-objectstore`](https://github.com/aiidateam/disk-objectstore)
+    implementation. If you ever wondered what the `_dos` appendix of the `core.psql_dos` and `core.sqlite_dos` storage
+    backends means, now you know! 😉
+
+[^2]:
+    While your workflows are being executed on the remote computer, files are located in a three level hierarchical
```

Comment:
I would make sure this does not become confusing - the paragraphs above all refer to internal AiiDA storage, here to the scratch.
E.g. start with something like "The discussion above refers to the files and data stored by AiiDA on the computer where AiiDA is installed, and that is preserved in the long term. These files are e.g. retrieved from the remote computer once a calculation finishes, or could e.g. be parsed data or input data provided by the user. Instead, while your workflows ..." (maybe replace workflows with calculations? only those are actually on the remote).

Maybe the sentence could be changed a bit along these lines:

Instead, the execution of calculations on the remote computer are preformed in a subfolder of the `work_directory` of the given `Computer` (typically in the `scratch`), where the subfolder name is generated from the UUID of the AiiDA `CalculationNode`. This directory has a three-level depth, obtained by "sharding" the UUID as follows: if the UUID is for instance `6861d8fb-4694-46be-b0e6-7282989f069d`, the calculation will run in a subfolder named `68/61/d8fb-4694-46be-b0e6-7282989f069d`.